File: src/main/java/me/soknight/minigram/chats/model/dto/ChatDto.java
package me.soknight.minigram.chats.model.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import me.soknight.minigram.chats.model.attribute.ChatType;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record ChatDto(
        @JsonProperty("id") long id,
        @JsonProperty("type") ChatType type,
        @JsonProperty("title") String title,
        @JsonProperty("owner_id") UUID ownerId,
        @JsonProperty("members") List<ChatMemberDto> members,
        @JsonProperty("last_message_id") Long lastMessageId,
        @JsonProperty("created_at") Instant createdAt,
        @JsonProperty("updated_at") Instant updatedAt
) {
}

File: src/main/kotlin/me/soknight/minigram/chats/dto/ChatDto.kt
@file:OptIn(ExperimentalUuidApi::class)

package me.soknight.minigram.chats.dto

import kotlinx.serialization.Serializable
import me.soknight.minigram.chats.model.ChatMemberRole
import me.soknight.minigram.chats.model.ChatType
import kotlin.time.Instant
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@Serializable
data class ChatDto(
    val id: Long,
    val type: ChatType,
    val title: String?,
    val ownerId: Uuid,
    val members: List<ChatMemberDto>,
    val lastMessageId: Long?,
    val createdAt: Instant,
    val updatedAt: Instant
)

@Serializable
data class ChatMemberDto(
    val profileId: Uuid,
    val name: String,
    val photoUrl: String?,
    val role: ChatMemberRole,
    val joinedAt: Instant
)

@Serializable
data class ChatMessageDto(
    val id: Long,
    val chat: ChatDto,
    val sender: ChatMemberDto,
    val content: String,
    val createdAt: Instant,
    val updatedAt: Instant
)
